pkg/kube/inject/inject.go

```diff
@@ -302,9 +302,17 @@ func injectRequired(ignored []string, config *Config, podSpec *corev1.PodSpec, m
 // ProxyImage constructs image url in a backwards compatible way.
 // values based name => {{ .Values.global.hub }}/{{ .Values.global.proxy.image }}:{{ .Values.global.tag }}
 func ProxyImage(values *opconfig.Values, image *proxyConfig.ProxyImage, annotations map[string]string) string {
-	imageName := "proxyv2"
 	global := values.GetGlobal()
 
+	imageName := "proxyv2"
+	if global.GetProxy() != nil && global.GetProxy().GetImage() != "" {
+		imageName = global.GetProxy().GetImage()
+		if strings.Contains(imageName, "/") {
```

Comment:
Would it be clearer to check if `imageName` is an image URL before calling the imageURL() function? Then, we wouldn't need to build it again and could simply return it.

Reply:
Not sure I follow - that is what this does - we call `imageURL()` on L330, while we check `imageName` on L310 and return early if it looks like a URL.

Do you mean checking in the template before calling `ProxyImage`? I think that's a bit more error prone due to the interaction between `proxy/proxy_init` both calling this logic, but I'm happy to update this to use that approach if that's what's preferred.

Reply:
More error prone in the sense that it pushes a lot of assumptions about the form the input takes outside the context of the function (e.g proxy.image must not be a URL if we're calling `ProxyImage`).